src/server/analytic_services/activity_log_store.js

```diff
@@ -36,26 +35,25 @@ class ActivityLogStore {
     }
 
 
-    create(activity_log) {
-        if (!activity_log._id) {
-            activity_log._id = this.make_activity_log_id();
-        }
-        return P.resolve()
-            .then(() => this._activitylogs.validate(activity_log))
-            .then(() => this._activitylogs.col().insertOne(activity_log))
-            .catch(err => mongo_utils.check_duplicate_key_conflict(err, 'audit_log'))
-            .return(activity_log);
+    async create(activity_log) {
+        return P.resolve().then(async () => {
```

Comment:
This would not work. Async function will convert anything to a native Promise, so doesn't matter what you return it won't be received by the caller as a Bluebird promise.
```javascript
$ node
Welcome to Node.js v12.18.4.
Type ".help" for more information.
> B = require('bluebird')
> f = async () => B.resolve(2)
[AsyncFunction: f]
> p = f()
Promise { <pending> }
> p instanceof B
false
> p instanceof Promise
true
```

Reply:
Missed removing async from `activity_log_store` and `history_data_store`.
Thank you.